lib/api/LogManagerImpl.cpp

```diff
@@ -804,15 +820,37 @@ namespace MAT_NS_BEGIN
         return m_dataViewerCollection;
     }
 
-    void LogManagerImpl::SetDataInspector(const std::shared_ptr<IDataInspector>& dataInspector)
+    void LogManagerImpl::AddOrUpdateInspector(const std::shared_ptr<IDataInspector>& dataInspector)
     {
         LOCKGUARD(m_dataInspectorGuard);
-        m_dataInspector = dataInspector;
+        if(dataInspector == nullptr)
+        {
+            m_dataInspectors.clear();
+            return;
+        }
+
+        auto itInspector = std::find_if(m_dataInspectors.begin(), m_dataInspectors.end(), [&dataInspector](const auto& inspector){
```

Comment:
"There can only be one of each kind and I find/replace it based on some key" sounds more like a map than a vector.

Reply:
Are there any performance differences to be considered that would favor `vector`? Because the `m_dataInspectors` vector will always be quite small